scripts/test-all.sh

```diff
@@ -33,25 +33,80 @@ run_test() {
     echo -e "${BLUE}Running $name...${NC}"
 
     if (cd "$(dirname "$0")/.." && eval "$command") > /tmp/test_output 2>&1; then
-        echo -e "${GREEN}✓ PASS${NC} - $name"
-        RESULTS+=("PASS: $name")
+        # Check if tests were skipped
+        skip_count=$(grep -c "^--- SKIP:" /tmp/test_output 2>/dev/null || true)
+        skip_count=${skip_count:-0}
```

Comment:
Performance impact is minimal. The grep operations are only performed on the already-captured test output in /tmp/test_output, not on live streams. The use of `|| true` prevents failures if no matches are found, and the default value assignment `${skip_count:-0}` is a good defensive programming practice.